service/storage_service.cc

```diff
@@ -1390,10 +1504,28 @@ std::unordered_set<locator::token> storage_service::get_tokens_for(inet_address
 }
 
 // Runs inside seastar::async context
-void storage_service::set_gossip_tokens(const std::unordered_set<dht::token>& local_tokens) {
+// Assumes that no other functions modify STREAMS_TIMESTAMP, TOKENS or STATUS
+// in the gossiper's local application state while this function runs.
+void storage_service::gossip_joined(
```

Comment:
gossip_joined is a bad name in my view. What does gossip joined mean? It suggests this node just finished joining the gossip which is not what this function does. 

 We can keep the old name. Yes, we now set more than tokens, but it does not matter, tokens are the most significant thing we set in this function.

Reply:
"gossip the fact that the node has joined"
how about "start_participating_in_the_ring"? (what the comment in `join_token_ring` says)

But I can leave the old name too (I don't like it though)

Reply:
For now I switched back to the old name